Allow options to be passed.  Allow { extname : false }

PrimeLens opened the pull request:
Allow options to be passed.    

Allow { extname : false }  to be passed in as options and this will prevent the filename extension appearing in the keys of the output.

Goal is to allow dot notation in referencing

briangonzalez commented on index.js:
```diff
@@ -41,6 +45,13 @@ module.exports = function (dest) {
       first = first || file;
       var id = file.path.replace(file.base, '').split('/').join(':');   // 'foo/bar/bax.txt' => 'foo:bar:baz.txt'
+      
+      if (options.extname===false) { 
```
spacing:

```
options.extname === false
```

briangonzalez commented on index.js:
```diff
@@ -9,8 +9,12 @@ nconf.file({ file: 'store.json' });
+module.exports = function (dest, options) {
+  
+  // set defaults for options here
+  options = options ? options : {};
```
```
options = options || {};
```

PrimeLens commented:
you got it. give me two secs

briangonzalez commented:
@PrimeLens Thanks for this! A few things - can you:
- Fix modifications in this PR.
- Squash all of these commits into a single commit (`git rebase -i`)
- Increment the version to `0.1.3` - https://github.com/briangonzalez/gulp-file-contents-to-json/blob/master/package.json#L3
- Write a test for the new functionality

briangonzalez commented:
PS - I can help with the test if you need guidance.

briangonzalez commented:
@PrimeLens Actually, scratch the test. I'll come back in and do that myself. Noticing coverage is abysmal  (see non-existent) on this project.

PrimeLens commented:
I'm cleaning the kids fish-tank at the moment, lol. i committed the space change just fro mthe browser but may not get to much else till I get a free second at work 2morra

briangonzalez commented:
Sounds good! Have fun cleaning the tank. Would love to know how you're using this little plugin.

PrimeLens commented:
I had a backbone/react boilerplate I built for macys that used grunt. Adapting it to gulp for macmillan.  This one will be used to gather all html partials and hang off window object. See this repo here https://github.com/PrimeLens/backbone-react-multipage-boilerplate-switchable old branch is the grunt build. the html partials are used on any page that is jquery only

PrimeLens commented:
it will take the place of grunt-htmlconvert

PrimeLens commented:
```
    .pipe(fc2json('htmlpartials.js'))
    .pipe(insert.prepend('window.htmlpartials = '))
    .pipe(insert.append(';'))
    .pipe(gulp.dest('./src/htmlpartials'));
```

PrimeLens commented:
I've updated the version number in package.json and fixed the spacing. I'm not sure how to rebase though.  Do I need to create a new pull request?

briangonzalez commented:
You'll need to squash these into a single commit then push to a new branch.

https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

PrimeLens commented:
There was a need in my use case to have a flat json output of only one layer deep based on filename. To implement one would need to remove the path. I added another option called 'flat'. Implementation was easy

``` javascript
      if (options.flat === true) { 
        // 'foo:bar:baz.txt' => 'baz.txt'
        // 'foo:bar:baz' => 'baz'
        id = id.split(':').reverse()[0]; 
      }; 
```

PrimeLens commented:
Hi Brian, with the help of some of the dev's on my team I've rebased into one commit. 
https://github.com/PrimeLens/gulp-file-contents-to-json/commits/master

briangonzalez commented:
@PrimeLens Beautiful! Thanks for contributing!

PrimeLens commented:
You're welcome.  When would it manifest itself in npm?  (I'm hoping to use that to propagate this change to my team)

briangonzalez commented:
1.3 has been published

https://www.npmjs.com/package/gulp-file-contents-to-json
